### PR TITLE
Rename TMC_CSP_TOKEN, CSP_ENDPOINT to VMW_CLOUD_ENDPOINT, VMW_CLOUD_API_TOKEN

### PR DESCRIPTION
**Pull request by shobha2626**
1. **What this PR does / why we need it**:

Rename TMC_CSP_TOKEN, CSP_ENDPOINT to VMW_CLOUD_ENDPOINT, VMW_CLOUD_API_TOKEN

2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

3. **Additional information**

Tested the provider with new env var

4. **Special notes for your reviewer**

### CODE REVIEW

**Review by Vasundharashukla — DISMISSED** [review]
LGTM

File: resource_templates/tmc_provider.tf [hunk 1]
```diff
@@ -14,5 +14,5 @@ terraform {
 // Basic details needed to configure Tanzu Mission Control provider
 provider "tmc" {
   endpoint = "<stack-name>.tmc-dev.cloud.vmware.com" // Required, environment variable: TMC_ENDPOINT
-  token    = "<csp-token>"                           // Required, environment variable: TMC_CSP_TOKEN
+  token    = "<vmw-cloud-api-token>"                 // Required, environment variable: VMW_CLOUD_API_TOKEN
```
Comment (shreyassreenivas) [inline resource_templates/tmc_provider.tf:17]: Even this should be updated to `vmw_cloud_api_token = "<vmw-cloud-api-token>" `

File: examples/provider/provider.tf [hunk 2]
```diff
@@ -1,8 +1,8 @@
 provider "tmc" {
   endpoint = var.endpoint # optionally use TMC_ENDPOINT env var
-  token = var.token # optionally use TMC_CSP_TOKEN env var
+  token = var.token # optionally use VMW_CLOUD_API_TOKEN env var
```
[the remark this answers is not in the document]
Comment (shobha2626) [reply examples/provider/provider.tf]: Yes, done

```
➜  usecases git:(mshobha/rename-env-var) ✗ tf apply
provider.tmc.endpoint
  Enter a value: mshobha.tmc-dev.cloud.vmware.com

provider.tmc.vmw_cloud_api_token
  Enter a value:
```

File: examples/provider/provider.tf [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
 
   # if you are using dev or different csp endpoint, change the default value below
   # for production environments the csp_endpoint is console.cloud.vmware.com
-  # csp_endpoint = "console.cloud.vmware.com" or optionally use TMC_CSP_ENDPOINT env var
+  # csp_endpoint = "console.cloud.vmware.com" or optionally use VMW_CLOUD_ENDPOINT env var
```
[the remark this answers is not in the document]
Comment (shobha2626) [reply examples/provider/provider.tf]: Done

File: examples/provider/provider.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (shreyassreenivas) [reply examples/provider/provider.tf]: The token should also be updated in the examples

File: resource_templates/tmc_provider.tf (hunk 1 above)
Comment (shreyassreenivas) [reply resource_templates/tmc_provider.tf]: the "token" string should also be changed

File: examples/provider/provider.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (shobha2626) [reply examples/provider/provider.tf]: Yes, this change is done previously. PTAL

File: resource_templates/tmc_provider.tf (hunk 1 above)
Comment (shobha2626) [reply resource_templates/tmc_provider.tf]: Yes, this change is done previously. PTAL

**Review by shreyassreenivas — APPROVED** [review]

**Review by Vasundharashukla — APPROVED** [review]

**Pull request merged**